modules/lib/core/index.d.ts

```diff
@@ -578,11 +578,12 @@ export interface PageComponent<
         Record<string, Region<(FragmentComponent | LayoutComponent | PartComponent | TextComponent)[]>> = 
         Record<string, Region<(FragmentComponent | Layout          | Part          | TextComponent)[]>>
 > {
-    type: 'page'
-    descriptor: Descriptor
-    config: Config
-    path: '/'
-    regions: Regions;
+    type?: 'page'
```

Comment:
What is the background of making every field optional here?

Reply:
I can make it better, gimme moment

Reply:
How bout dat?